pkg/name/digest_test.go

```diff
@@ -38,6 +38,7 @@ var goodStrictValidationTagDigestNames = []string{
 var goodWeakValidationDigestNames = []string{
 	"namespace/pathcomponent/image@" + validDigest,
 	"library/ubuntu@" + validDigest,
+	"gcr.io/CROSSplane/PROVIDER-gcp@" + validDigest,
```

Comment:
Do we care to transform and test `GCR.io` as well?

Reply:
Given that domain names are case-insensitive, I could be convinced that "we don't need to" but also "we can if we want".

Reply:
DNS names are case insensitive, so technically we could, and this would be fine.  I guess I'd be more interested in what the Docker behavior here is, and matching that.  I am in a meeting now and don't have a docker setup, so if you are able to quickly tell me what it does I'll patch after I get out.  👍